yelp_bytes.py

```diff
@@ -26,22 +37,24 @@ def from_bytes(obj, encoding='internet', errors='strict'):
 
     This function always returns unicode.
     """
-    if isinstance(obj, unicode):
+    mro = type(obj).mro()
+    if unicode in mro:
         return obj
+    elif bytes in mro:
+        return obj.decode(encoding, errors)
+    elif hasattr(obj, '__bytes__'):
+        return obj.__bytes__().decode(encoding, errors)
```

Comment:
It's weird, but comment out that line and see the tests fail

The function is called from_bytes, and if __bytes__ is defined, it would be the canonical data source, is my rationalization.